### PR TITLE
Harmonize forms

### PR DESCRIPTION
**Pull request by alanthai**
Added session, login/logout, form validation

### CODE REVIEW

File: src/containers/counter-page.ts [hunk 1]
```diff
@@ -5,34 +7,31 @@ export class RioCounterPage {
...
+    @Inject('$rootScope') $rootScope) {
 
-  constructor($ngRedux, $scope: ng.IScope, $rootScope) {
     const disconnect = $ngRedux.connect(
       this.mapStateToThis, CounterActions)(this);
 
     // Needed for redux devtools to be able to update application state.
     const unsubscribe = $ngRedux.subscribe(_ => {
-      setTimeout($rootScope.$apply, 100);
+      setTimeout($rootScope.$apply.bind($rootScope), 100);
```
Comment (alanthai) [inline src/containers/counter-page.ts]: Quick fix of error clicking increment/decrement buttons.

File: src/components/index.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
-export * from './counter';
+export * from './alert';
 export * from './button';
-export * from './navigator/navigator';
-export * from './navigator/navigator-item';
+export * from './counter';
+export * from './form';
+export * from './input';
+export * from './login';
+export * from './input';
```
Comment (SethDavenport) [inline src/components/index.ts]: Looks like you have a duplicate line here (for './input')

File: src/components/logo/index.ts [hunk 3]
```diff
@@ -5,12 +5,12 @@ export class RioLogo {
   static selector = 'rioLogo';
 
   static options: ng.IComponentOptions = {
-    controllerAs: 'ctrl',
+    controllerAs: 'logoCtrl',
```
Comment (SethDavenport) [inline src/components/logo/index.ts]: +1 for giving the controllers specific names.  Makes it easier to understand where the bindings live when looking at the top-level template.

File: src/containers/app.ts [hunk 4]
```diff
@@ -1,27 +1,92 @@
...
     controller: RioApp,
+    controllerAs: 'appCtrl',
     template: `
       <div class="col-12">
+        <rio-login-modal
+          on-submit="appCtrl.login(credentials)"
+          has-error="appCtrl.session.get('hasError', false)"
+          is-pending="appCtrl.session.get('isLoading', false)"
+          ng-if="!appCtrl.isLoggedIn"></login-modal
+        ></rio-login-modal>
```
Comment (SethDavenport) [inline src/containers/app.ts]: Looks like there's an extra closing tag in here somehow.

File: src/containers/app.ts [hunk 5]
```diff
@@ -1,27 +1,92 @@
...
+  static $inject = [
+    '$ngRedux',
+    '$scope'
+  ];
+
+  constructor($ngRedux, $scope: ng.IScope) {
+    const unsubscribe = $ngRedux.connect(
+      this.mapStateToThis, this.mapDispatchToThis)(this);
+
+    $scope.$on('$destroy', unsubscribe);
```
Comment (SethDavenport) [inline src/containers/app.ts]: :+1: for remembering to unsubscribe

File: src/containers/counter-page.ts (hunk 1 above)
Comment (SethDavenport) [inline src/containers/counter-page.ts]: I'd like to understand this more - what was the error?

File: src/index.html [hunk 6]
```diff
@@ -6,6 +6,7 @@
     <meta name="description" content="">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>Rangle.io - Angular / Redux seed</title>
+    <base href="/">
```
Comment (SethDavenport) [inline src/index.html]: :+1:

File: src/store/configure-store.ts [hunk 7]
```diff
@@ -2,27 +2,26 @@
 
 import ReduxThunk from 'redux-thunk';
```
Comment (SethDavenport) [inline src/store/configure-store.ts]: This is missing the persist-state stuff: https://github.com/rangle/angular2-redux-starter/blob/master/src/store/configure-store.ts

The other starters use it to remember that you've logged in when you refresh the browser.

File: src/containers/counter-page.ts (hunk 1 above)
Comment (alanthai) [inline src/containers/counter-page.ts]: Here's the console error
```
TypeError: this.$eval is not a function
    at Scope.$apply (angular.js:16984)
```

Comment (SethDavenport) [inline src/containers/counter-page.ts]: OK yeah thanks.

**Comment by SethDavenport** [comment]
:shipit:

**Pull request merged**